### PR TITLE
Express server strict mode

### PR DESCRIPTION
**Pull request by Brandon-Perry**
**1. What is the goal of this PR?**
Make it so we're using strict:true on the express-server, which makes type-checking stronger 
**2. What specific parts of T3C are you changing and how?**
Changed ts-config in express/common to use strict:true, changed places in code that started to throw type errors.

Note: in some places I used ts-ignore, which is generally not good practice, but those places desperately need refactor / would have been a headache to do. 
**3. How did you test these changes?**
Ran pets

Please add one or more reviewer(s) and tag them in a new post in the Slack dev channel :)

### CODE REVIEW

File: express-server/src/googlesheet.ts [hunk 1]
```diff
@@ -63,7 +64,11 @@ export async function fetchSpreadsheetData(
   const emailToData: { [key: number]: { id: string; comment: string }[] } = {};
 
   rows.forEach((row, id) => {
+    // ! Brandon: indexing into a string and then using that to access a record with key:number wtf?!?
+    // ! Brandon: Adding ts ignores into this so we can use strict mode and not break shit. This entire file needs redone.
```
Comment (staceysv) [inline express-server/src/googlesheet.ts:68]: really appreciate the comment & understand the feeling!
nit/general reminder, our repo is open source, let's keep the language more professional
e.g. "access a record with key:number — TODO: investigate why/how"
e.g. "so we can use strict mode and not break anything — TODO: redo this entire file" (added to task tracking btw)

File: express-server/src/pipeline.ts [hunk 2]
```diff
@@ -192,7 +194,7 @@ async function pipeline(
       : `${secs} seconds`;
 
   // next line is important to avoid leaking keys!
-  delete options.apiKey;
+  options.apiKey = "";
```
Comment (staceysv) [inline express-server/src/pipeline.ts:197]: general: are we using this file at all? I thought we're only using workers.ts for the pipeline these days? can we get rid of this whole file if yes?

File: express-server/src/routes/report.ts [hunk 3]
```diff
@@ -7,6 +7,10 @@ export async function report(req: Request, res: Response) {
     api.getReportRequestUri.parse(req.params.reportUri),
   );
   const name = new URL(uri).pathname.split("/").pop();
+  if (!name) {
```
Comment (staceysv) [inline express-server/src/routes/report.ts:10]: can we add a comment on what this state is/what this means? and/or log what this means — malformed request or bad input url / unable to extract report name?

**Review by staceysv — CHANGES_REQUESTED** [review]
Looks good, thanks for this cleanup! Left a few minor comments in places I would like a bit more explanation or see an opportunity for more cleanup (e.g. deleting deprecated files).
Q1: shall we test this a bit more before merging to prod? Any edge cases you can anticipate?
Q2: Also, how temporary do you think these typescript workarounds are/how urgent is it to refactor?

File: express-server/src/pipeline.ts (hunk 2 above)
Comment (Brandon-Perry) [reply express-server/src/pipeline.ts:197]: will remove in a future PR

File: express-server/src/routes/generate.ts [hunk 4]
```diff
@@ -16,13 +16,20 @@ export default async function generate(req: Request, res: Response) {
   try {
     const body: GenerateApiRequest = req.body;
     const { data: dataPayload, userConfig } = body;
```
[the remark this answers is not in the document]
Comment (Brandon-Perry) [reply express-server/src/routes/generate.ts:18]: Will remove in a future PR

**Comment by Brandon-Perry** [comment]
As far as testing goes - I think it's fine. There are no real changes to the logic, it's all type changes. If there as an issue with it, it would have thrown while transpiling. 

So I do think the backend needs a pretty extensive refactor. The issue with ts-ignore (which is used a lot in this PR) is that we lose the benefit of TypeScript in that spot. So if, for example, we change a type, it could mean that there's an error there that we're not catching. The issue however, is that the way that the backend is currently written would mean that more extensive changes would need to be made.
In terms of priority, I definitely say it should be a priority. But, that doesn't necessarily mean that we drop everything and rewrite it. I think doing so in pieces is more maintainable. So as we're implementing new features (like the private / public buckets), we should be refactoring as we go, instead of just trying to jerry-rig it.

**Comment by staceysv** [comment]
Beautiful, thanks for all the responses & comments :) — let's make sure to keep refactoring as we go!

**Review by staceysv — APPROVED** [review]
